### PR TITLE
Simplify toplevel cmake

### PR DESCRIPTION
**Pull request by alain-rs**
> ### Motivate of the pull request
> - [ x] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

Push all Tcl build recipe under third_party/tcl_cmake

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -179,64 +179,37 @@ if (MSVC)
...
-    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/third_party/tcl/unix"
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/tcl/generic/tcl.h
-  )
-  add_custom_target(tcl_stubb_build DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/lib/${TCL_STUBB_LIB})
-  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lib/${TCL_STUBB_LIB}
-    COMMAND echo "       Compiling Tcl Stubb"
-    COMMAND echo "       Tcl Stubb Compilation completed"
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/tcl/generic/tcl.h
-  )
+  ADD_LIBRARY(tcl_static STATIC IMPORTED)
```
Comment (KochynVolodymyr) [inline CMakeLists.txt:204]: can you do it in lower case like all other cmake functions?

File: CMakeLists.txt [hunk 2]
```diff
@@ -179,64 +179,37 @@ if (MSVC)
...
-    COMMAND echo "       Tcl Compilation completed"
-    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/third_party/tcl/win"
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/tcl/generic/tcl.h
-  )
-  # Copy the tcl lib from source to build directory
-  add_custom_command(TARGET tcl_build POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy 
-            ${CMAKE_CURRENT_SOURCE_DIR}/third_party/tcl/win/Release_AMD64_VC1929/${TCL_STATIC_LIB}
-            ${CMAKE_CURRENT_BINARY_DIR}/lib/foedag/lib/${TCL_STATIC_LIB}
+  ADD_LIBRARY(tcl_static STATIC IMPORTED)
```
Comment (KochynVolodymyr) [inline CMakeLists.txt:182]: This:

ADD_LIBRARY(tcl_static STATIC IMPORTED)
  SET_TARGET_PROPERTIES(tcl_static PROPERTIES
    IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/lib/${TCL_STATIC_LIB})

  ADD_LIBRARY(tcl_stubb STATIC IMPORTED)
  SET_TARGET_PROPERTIES(tcl_stubb PROPERTIES
    IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/lib/${TCL_STUBB_LIB})

coule be moved outside if else (same code)

File: CMakeLists.txt (hunk 1 above)
Comment (alain-rs) [reply CMakeLists.txt:204]: Will address in a separate pull-request

File: CMakeLists.txt (hunk 2 above)
Comment (alain-rs) [reply CMakeLists.txt:182]: Will address in a separate pull-request

**Pull request merged**